lib/astutils.cpp

```diff
@@ -1329,7 +1335,7 @@ bool isSameExpression(bool cpp, bool macro, const Token *tok1, const Token *tok2
         const Token *end1 = t1->link();
         const Token *end2 = t2->link();
         while (t1 && t2 && t1 != end1 && t2 != end2) {
-            if (t1->str() != t2->str() || t1->flags() != t2->flags())
+            if (t1->str() != t2->str() || flagsDiffer(t1, t2, true))
```

Comment:
question shouldn't the `macro` be used here?

Reply:
I wasn't sure what `macro` refers to, the complete expression, or also parts of it? At least it's consistent now.

Reply:
:+1: 
yes I think it should be effective for parts of it too.